src/share.coffee

```diff
@@ -50,10 +50,27 @@ class Share extends ShareUtils
 
     return @
 
+  isNodeList: (nodes) ->
+    stringRepr = Object.prototype.toString.call(nodes)
+
+    return typeof nodes is 'object' and
+        /^\[object (HTMLCollection|NodeList|Object)\]$/.test(stringRepr) and
+        nodes.hasOwnProperty('length') and
+        (nodes.length == 0 || (typeof nodes[0] is "object" and nodes[0].nodeType > 0))
+
+  isNode: (node) ->
+    return node and node.nodeType and
+      (node.nodeType == 1 || node.nodeType == 11)
 
   setup: (element, opts) ->
     ## Record all instances
-    instances = document.querySelectorAll(element) # TODO: Use more efficient method.
+
+    if @isNodeList(element)
```

Comment:
I'm not fluent with Coffescript actually :)

Anyway I don't like this version, I find the one I did more readable